### PR TITLE
Linters

### PR DESCRIPTION
**Pull request by DeVuDeveloper**
**Hi,
Adding locally and GitHub linters.**

## Set up GitHub Actions

This GitHub Action is going to run [Rubocop](https://docs.rubocop.org/en/stable/) and [Stylelint](https://stylelint.io/) to help to find style issues.

[Stylelint](https://stylelint.io/) is a linter for your stylesheets that helps you avoid errors and enforce conventions.

1. Created a `.github/workflows` folder and added / linters.yml to that folder.
2. When we open your first pull request we should see the result of the GitHub Actions.

### Rubocop

1. Added this line to the `Gemfile`
    ```
    gem 'rubocop', '>= 1.0', '< 2.0'
    ```
2. Run `bundle install`.
3. Added  [.rubocop.yml](./.rubocop.yml) to the root directory.
4. Run `rubocop`.
5. Fix linter errors.
6. In auto-correct mode, RuboCop will try to automatically fix offenses:

$ rubocop -A
# or
$ rubocop --auto-correct-all

### Stylelint

1. Run

   ```
   npm install --save-dev stylelint@13.x stylelint-scss@3.x stylelint-config-standard@21.x stylelint-csstree-validator@1.x
   ```

2. Added [.stylelintrc.json](./.stylelintrc.json) to the root directory.

3. Run `npx stylelint "**/*.{css,scss}"` on the root of project directory.
5. Fix linter errors.

 **Important Note: Linter prefers a single quote to use in the project.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -28,3 +28,4 @@
 !/app/assets/builds/.keep
 
 /node_modules
+/Gemfile
```
Comment (duleorlovic) [inline .gitignore:31]: Why we need to ignore `Gemfile` ?

File: .rubocop.yml [hunk 2]
```diff
@@ -0,0 +1,60 @@
+AllCops:
```
Comment (duleorlovic) [inline .rubocop.yml:1]: Can you put in a comment as a first line, where did you find this configuration ?

File: .ruby-version [hunk 3]
```diff
@@ -1 +1 @@
-3.2.0
+3.1.2
```
Comment (duleorlovic) [inline .ruby-version:1]: Can we use latest versions  ? If you have problems installing, checkout https://dev.trk.in.rs/2023/01/11/install-ruby-using-brew-rbenv.html#installing-ruby-using-rbenv

File: app/javascript/controllers/dist/hello_controller.dev.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
+"use strict";
```
Comment (duleorlovic) [inline app/javascript/controllers/dist/hello_controller.dev.js:1]: Can you add stimulus in another task ? 
This PR is for linters

**Comment by DeVuDeveloper** [comment]
Ok, I am back, I will check all comments and back to you.

File: .gitignore (hunk 1 above)
Comment (DeVuDeveloper) [reply .gitignore:31]: We need both Gemfile and Gemfile.lock
I checked if it came from a branch before, but not.
My mistake, sorry.

File: .rubocop.yml (hunk 2 above)
Comment (DeVuDeveloper) [reply .rubocop.yml:1]: I worked for a US company where we used this configuration and it works well, but on PR, not push what I find unnecessary.
For tests, I wait for the app to grow and to see what tests we use. Btw I never used more than the configuration  I made.
Repo with linter is private and I can't show the source, sorry.

Comment (DeVuDeveloper) [reply .rubocop.yml:1]: Ok, it worked both locally and for GitHub. With the correction of errors easily locally, GitHub was to pass the linter test.
Now I will delete the branch and then Pr will be on the new one.

**Pull request closed without merging**